src/api/forms/service.js

```diff
@@ -133,6 +133,9 @@ export async function updateDraftFormDefinition(formId, definition, author) {
       throw Boom.badRequest(`Form with ID '${formId}' has no draft state`)
     }
 
+    // patch the form definition with attributes from the title, where they're not designed to be customised
+    definition.name = form.title
```

Comment:
Should we guard it so it only overwrites `name` when nullish?

```suggestion
    definition.name ??= form.title
```

Reply:
These should always match, so we want to always overwrite with the latest correct value from the metadata. Users shouldn't be able to customise this (confirmed by Naz).

Reply:
Yeah, so if it's already set then it already matches?

Reply:
Not if the user has changed it by downloading the form and tweaking the name manually, this protects against that too.

Reply:
e.g. if I upload

```json
{
    name: "a title that shouldn't be here"
    pages: []
}
```

then `definition.name ??= form.title` wouldn't overwrite it with the value from the metadata.

Reply:
Ah sorry, didn't know about this requirement

Otherwise it's rewriting the same property over and over, if that's alright?

Reply:
Yeah that's what I expect, won't waste much computation as we're already fetching the form metadata. Cheers.